tests/controller/util.test.ts

```diff
@@ -0,0 +1,110 @@
+      totalSolAmount: 0.1,
+      totalSolCost: "0.100000001",
+      totalSolFee: 1e-9,
+      transactionType: "",
+    };
+    assert.deepEqual(result, assertResult);
+  });
+  it("calculateTxFee", async () => {
+    const tx = new Transaction({ recentBlockhash: sKeyPair[0].publicKey.toBase58(), feePayer: sKeyPair[0].publicKey });
+    tx.add(transferInstruction());
+    const result = await solanaHelper.calculateTxFee(tx.compileMessage(), mockGetConnection());
+    assert.deepEqual(result.fee, 1);
+  });
+  // it("getEstimateBalanceChange", async () => {
+  //   const tx = new Transaction({ recentBlockhash: sKeyPair[0].publicKey.toBase58(), feePayer: sKeyPair[0].publicKey });
+  //   tx.add(transferInstruction());
+  //   const result = await solanaHelper.getEstimateBalanceChange(mockGetConnection(), tx, sKeyPair[0].publicKey.toBase58());
+  //   assert.deepEqual(result, [
+  //     {
+  //       address: "7dpVde1yJCzpz2bKNiXWh7sBJk7PFvv576HnyFCrgNyW",
+  //       changes: 0,
+  //       decimals: 9,
+  //       mint: "",
+  //       symbol: "SOL",
+  //     },
+  //   ]);
+  // });
+  // it("calculateChanges", async () => {
+  //   const tx = new Transaction({ recentBlockhash: sKeyPair[0].publicKey.toBase58(), feePayer: sKeyPair[0].publicKey });
+  //   tx.add(transferInstruction());
+  //   const result = await solanaHelper.calculateChanges(mockGetConnection(), mockSimulateTransaction, "7dpVde1yJCzpz2bKNiXWh7sBJk7PFvv576HnyFCrgNyW", [
+  //     "7dpVde1yJCzpz2bKNiXWh7sBJk7PFvv576HnyFCrgNyW",
+  //     "6bS8uykyBg1dC5E4fatWaJC177KTcyW5GsGtUvNq3RPz",
+  //   ]);
+  //   assert.deepEqual(result, [
+  //     {
+  //       address: "7dpVde1yJCzpz2bKNiXWh7sBJk7PFvv576HnyFCrgNyW",
+  //       changes: 0,
+  //       decimals: 9,
+  //       mint: "",
+  //       symbol: "SOL",
+  //     },
+  //   ]);
+  // });
+  it("getClubbedNfts", async () => {
+    const result = await solanaHelper.getClubbedNfts(mockNFTs);
+    assert.deepEqual(result, mockClubbedNFTs);
+  });
+  it("ruleVerifierId", () => {
+    const solResult = solanaHelper.ruleVerifierId("sol", sKeyPair[0].publicKey.toBase58());
+    assert.deepEqual(solResult, true);
+    const googleResult = solanaHelper.ruleVerifierId("google", "test@gmail.com");
+    assert.deepEqual(googleResult, true);
+    const redditResult = solanaHelper.ruleVerifierId("reddit", "test123");
+    assert.deepEqual(redditResult, true);
+    const twitterResult = solanaHelper.ruleVerifierId("twitter", "testweb3");
+    assert.deepEqual(twitterResult, true);
+    const githubResult = solanaHelper.ruleVerifierId("github", "testweb3");
+    assert.deepEqual(githubResult, true);
+  });
+  it("generateSPLTransaction", async () => {
+    const result = await solanaHelper.generateSPLTransaction(
+      sKeyPair[1].publicKey.toBase58(),
+      1,
+      mockNFTs[0],
+      sKeyPair[0].publicKey.toBase58(),
+      mockGetConnection()
+    );
+    assert.deepEqual(result.feePayer, sKeyPair[0].publicKey);
```

Comment:
Could add checking and cases where the receiver is sol account address and token account address

Reply:
Also case where create Account Instruction is needed